### ISSUE TITLE
RCM_MsgTypes_Pkg undefined

### ISSUE THREAD

**Issue by jokaICS** [openETCS/modeling#730]
@UweSteinkeFromSiemens 
After your commit https://github.com/openETCS/modeling/commit/a2da4e770081cf5bef0c22c64e0620f555d2abc5, I have some error messages regarding MoRC:

```
Semantic Error ERR_7 Unknown package at MoRC_Pck::MoRC_Main/inMessage_new/ 
the package RCM_MsgTypes_Pkg:: is undefined 
Semantic Error ERR_7 Unknown package at MoRC_Pck::MoRC_Main/_L195/ 
the package RCM_MsgTypes_Pkg:: is undefined 
Semantic Error ERR_7 Unknown package at MoRC_Pck::Coder_Pkg::decM32/inMessage/ 
the package RCM_MsgTypes_Pkg:: is undefined 
Semantic Error ERR_7 Unknown package at MoRC_Pck::Coder_Pkg::decM32/_L1/ 
the package RCM_MsgTypes_Pkg:: is undefined 
Semantic Error ERR_7 Unknown package at MoRC_Pck::Coder_Pkg::decM38/inMessage/ 
the package RCM_MsgTypes_Pkg:: is undefined 
Semantic Error ERR_7 Unknown package at MoRC_Pck::Coder_Pkg::decM38/_L1/ 
the package RCM_MsgTypes_Pkg:: is undefined 
Semantic Error ERR_7 Unknown package at MoRC_Pck::Coder_Pkg::decM39/inMessage/ 
the package RCM_MsgTypes_Pkg:: is undefined 
Semantic Error ERR_7 Unknown package at MoRC_Pck::Coder_Pkg::decM39/_L1/ 
the package RCM_MsgTypes_Pkg:: is undefined 
Semantic Error ERR_7 Unknown package at MoRC_Pck::Coder_Pkg::decP42/inMessage/ 
the package RCM_MsgTypes_Pkg:: is undefined 
Semantic Error ERR_7 Unknown package at MoRC_Pck::Coder_Pkg::decP42/_L1/ 
the package RCM_MsgTypes_Pkg:: is undefined 
Semantic Error ERR_7 Unknown package at MoRC_Pck::Coder_Pkg::decP45/inMessage/ 
the package RCM_MsgTypes_Pkg:: is undefined 
Semantic Error ERR_7 Unknown package at MoRC_Pck::Coder_Pkg::decP45/_L1/ 
the package RCM_MsgTypes_Pkg:: is undefined 
```

Do we need to change some .etp files?

**Comment by UweSteinkeFromSiemens** [comment]
No. The reason is, that some of the files I pushed to github seem not be stored there, although they are marked as versioned and uploaded in my local clone. Currently I don't have an explanation for this misbehavior. Might be a bug of TortoiseGit or github. 

I will try to fix it.

**Comment by UweSteinkeFromSiemens** [comment]
@jokaICS: Please check, if 4 files in the directory \modeling\model\Scade\System\ObuFunctions\Radio\RCM_Types of your local clone are available.

**Comment by jokaICS** [comment]
> Please check, if 4 files in the directory \modeling\model\Scade\System\ObuFunctions\Radio\RCM_Types of your local clone are available.

On the master? I don't see a folder RCM_Types in https://github.com/openETCS/modeling/tree/master/model/Scade/System/ObuFunctions/Radio

**Comment by UweSteinkeFromSiemens** [comment]
Yes, on the master. This is, what confuses me too: looking at the repository with the browser does not display the directory. But I cloned the repository fresh from github, and the folder and the files included were cloned properly. It's unbelievable.

**Comment by jokaICS** [comment]
Strange indeed... anyway: I've pulled the master, but the files aren't there. Maybe deleting + re-adding the files will help?

**Comment by UweSteinkeFromSiemens** [comment]
I just tried exactly this, as you can see in the attached screen shot. But without success, the files are not displayed in the browser afterwards. Nevertheless, as the fresh clone proved, they are available. 
Probably, we should ask if not only you is affected by this problem. 
  
![screenshot](https://cloud.githubusercontent.com/assets/2225951/9902569/bcf0f306-5c6f-11e5-9d67-3f6eb52a7f1b.png)

@stefan-karg: could you perhaps try if you are able to pull these files from github ?

**Comment by jokaICS** [comment]
Hmm, I've created a fresh clone of the repo, but I still don't see a folder `RMC_Types` in `Radio`.

**Comment by apinnau** [comment]
Just pulled from master repo into a clean working directory **and the folder RCM_Types is there containing 4 files.**

**Comment by jokaICS** [comment]
@pinnau maybe I'm looking in the wrong place... could you please give me the github link to this folder?

**Comment by UweSteinkeFromSiemens** [comment]
Puuh, ... sounds well. I have an additional cloning process running. 

@jokaICS: Can it be perhaps the path length, that raises the problem?

**Comment by apinnau** [comment]
The full path is: **model\Scade\System\ObuFunctions\Radio\RCM_Types**
By the way: I'm using TortoiseGit as Uwe, too.

**Comment by jokaICS** [comment]
@pinnau I don't mean the file path, but its github path (web URL)

**Comment by jokaICS** [comment]
> @jokaICS: Can it be perhaps the path length, that raises the problem?

hmm, shouldn't be -- there are files with a far longer path.

**Comment by apinnau** [comment]
@jokaICS As Uwe already mentioned is the directory not visible in the web-view. It's the same for me... folder is visible in fresh synced working directory but not on github webservice via browser.

**Comment by UweSteinkeFromSiemens** [comment]
My additonal virgin clone is done. It comprises the questioned files.

**Comment by jokaICS** [comment]
Ok, then there is obviously some special Tortoise behavior; I've never seen, that there are "hidden" files that won't show up on github.

My proposal to solve this issue: @UweSteinkeFromSiemens could you please send me your files by mail, and then I will try to add these to my fresh clone. If they are already there, git should refuse the push.

**Comment by UweSteinkeFromSiemens** [comment]
@jokaICS: Could you please send me your email address => uwe.steinke@siemens.com

**Comment by jokaICS** [comment]
OK, I've pushed the files provided by @UweSteinkeFromSiemens, could please pull and to verify?

I presume that this issue is related to the case sensitivity (or insensitivity) of the involved file systems. the pull message shows, that the files in question have been renamed from `Radio/RCM_Type` to `radio/RCM_Types`. AFAIK, `radio` and `Radio` represent the same folder on windows...

**Comment by UweSteinkeFromSiemens** [comment]
I've pulled the files, and theý are available as they have been before. The log says, that you added these files. 
Very strange: the log tells, that the rename ocuured automatically between the 2 commits, when I added these files to version control after removing them before (without deleting them locally). I did not touch the directory "R(r)adio" at all. 
And this rename occured during my tries to investigate the effect this morning. So it cannot solve as an explanation, since you missed the files before already. 

Let us enjoy the mysticism of git, github, TortioseGit and Windows ... :100:
